src/ring_theory/dedekind_domain/ideal.lean

```diff
@@ -730,3 +731,47 @@ begin
 end
 
 end is_dedekind_domain
+
+section height_one_spectrum
+
+/-!
+### Height one spectrum of a Dedekind domain
+If `R` is a Dedekind domain of Krull dimension 1, the maximal ideals of `R` are exactly its nonzero
+prime ideals.
+We define `height_one_spectrum` and provide lemmas to recover the facts that prime ideals of height
+one are prime and irreducible. -/
+
+namespace is_dedekind_domain
+
+variables [is_domain R] [is_dedekind_domain R]
+
+/-- The height one prime spectrum of a Dedekind domain `R` is the type of nonzero prime ideals of
+`R`. Note that this equals the maximal spectrum if `R` has Krull dimension 1. -/
+@[nolint has_inhabited_instance unused_arguments]
+structure height_one_spectrum :=
+(as_ideal : ideal R)
+(is_prime : as_ideal.is_prime)
+(ne_bot   : as_ideal ≠ ⊥)
+
+variables (v : height_one_spectrum R) {R}
+
+lemma ideal.prime_of_height_one (v : height_one_spectrum R) : prime v.as_ideal :=
+ideal.prime_of_is_prime v.ne_bot v.is_prime
+
+lemma ideal.irreducible_of_height_one (v : height_one_spectrum R) :
+  irreducible v.as_ideal :=
+begin
+  rw [unique_factorization_monoid.irreducible_iff_prime],
+  apply ideal.prime_of_height_one v,
+end
+
+lemma associates.irreducible_of_height_one (v : height_one_spectrum R) :
+  irreducible (associates.mk v.as_ideal) :=
+begin
+  rw [associates.irreducible_mk _],
+  apply ideal.irreducible_of_height_one v,
+end
```

Comment:
Let's use dot notation here, otherwise the full names would look like `is_dedekind_domain.associates.irreducible_of_height_one`.
```suggestion
lemma height_one_spectrum.prime (v : height_one_spectrum R) : prime v.as_ideal :=
ideal.prime_of_is_prime v.ne_bot v.is_prime

lemma height_one_spectrum.irreducible (v : height_one_spectrum R) :
  irreducible v.as_ideal :=
unique_factorization_monoid.irreducible_iff_prime.mpr v.prime

lemma height_one_spectrum (v : height_one_spectrum R) :
  irreducible (associates.mk v.as_ideal) :=
(associates.irreducible_mk _).mpr v.irreducible
```

Reply:
Thanks, I'll make these changes.

Reply:
Actually, I will change the names but I'm not sure about the proofs - I wrote them with `rw - apply` because this makes them faster.